src/condor_schedd.V6/schedd.cpp

```diff
@@ -7686,6 +7686,75 @@ Scheduler::negotiate(int command, Stream* s)
 }
 
 
+int
+Scheduler::CmdDirectAttach(int, Stream* stream)
+{
+	ReliSock *rsock = (ReliSock*)stream;
+	ClassAd cmd_ad;
+	int num_ads = 0;
+	std::string claim_id;
+	ClassAd slot_ad;
+	std::string slot_user;
+	std::string slot_submitter;
+	PROC_ID jobid;
+	jobid.cluster = jobid.proc = -1;
+
+	dprintf(D_ALWAYS, "Got DIRECT_ATTACH from %s\n", rsock->peer_description());
+
+	if (!getClassAd(rsock, cmd_ad)) {
+		dprintf(D_ALWAYS, "CmdDirectAttach() failed to read command ad\n");
+		return 0;
+	}
+
+	slot_user = rsock->getFullyQualifiedUser();
+
+	if (!cmd_ad.LookupString(ATTR_SUBMITTER, slot_submitter)) {
+		slot_submitter = slot_user;
+	}
+
+		// TODO handle alternate submitter names
+	MainScheddNegotiate sn(0, nullptr, slot_submitter.c_str(), nullptr);
+
+	cmd_ad.LookupInteger(ATTR_NUM_ADS, num_ads);
+	dprintf(D_ALWAYS, "CmdDirectAttach() reading %d slot ads\n", num_ads);
+
+	for (int i = 0; i < num_ads; i++) {
+		std::string slot_name;
+		if (!rsock->get_secret(claim_id) || !getClassAd(rsock, slot_ad)) {
+			dprintf(D_ALWAYS, "CmdDirectAttach() failed to read slot ad %d\n", i);
+			return 0;
+		}
+
+			// TODO allow trusted users to match all jobs
```

Comment:
This would be nice for the AWS annex, but as such, is not a high priority.

Reply:
Leaving this as future work.